### DISCUSSION TITLE
OME Rest API SDK

### DISCUSSION THREAD

**Discussion by nums** [OvenMediaLabs/OvenMediaEngine#1349] [Discover]
Hello, 

I've started the development of a TypeScript SDK to utilize the Rest API of OvenMediaEngine. I've just begun, but I plan to enhance it with all the available endpoints. Please feel free to provide feedback / PR :
https://github.com/nums/ovenmediaengine-ts-sdk

I'd also like to try generating this same type of SDK for different languages. With artificial intelligence, this could be somewhat automated. Do you know of any tools that would allow this?

**Comment by krisdahl** [comment]
Nice, I think I’ll use it!

**Comment by nums** [comment]
What do you think about using OpenAPI specification to describe OME API ?

After a little search, I've found this service https://speakeasyapi.dev/ that can generate SDK in many languages.
